Apps/W1/HybridGP/app/src/Migration/History/GPPopulateHistTables.Codeunit.al

```diff
@@ -0,0 +1,957 @@
+            if GPCompanyAdditionalSettings.GetMigrateHistGLTrx() then begin
+                HistMigrationStatusMgmt.UpdateStepStatus("Hist. Migration Step Type"::"GP GL Journal Trx.", false);
+                PopulateOpenYearPostedTransactions();
+                PopulateHistoricalYearPostedTransactions();
+                HistMigrationStatusMgmt.UpdateStepStatus("Hist. Migration Step Type"::"GP GL Journal Trx.", true);
+            end;
+    end;
+
+    local procedure PopulateHistGLAccounts()
+    var
+        GPGL00105: Record "GP GL00105";
+        GPGL00100: Record "GP GL00100";
+        HistGLAccount: Record "Hist. G/L Account";
+    begin
+        if GPGL00105.FindSet() then
+            repeat
+                Clear(HistGLAccount);
+                if GPGL00100.Get(GPGL00105.ACTINDX) then begin
+                    HistGLAccount."No." := CopyStr(Format(GPGL00105.ACTNUMST), 1, MaxStrLen(HistGLAccount."No."));
+                    HistGLAccount.Name := GPGL00100.ACTDESCR;
+
+                    if not HistGLAccount.Insert() then
+                        HistMigrationStatusMgmt.ReportError("Hist. Migration Step Type"::"GP GL Accounts", GPGL00105.ACTNUMST,
+                            GetLastErrorCode(), GetLastErrorText());
+                end;
+            until GPGL00105.Next() = 0;
+    end;
+
+    local procedure PopulateOpenYearPostedTransactions()
+    var
+        GPGL20000: Record "GP GL20000";
+        GPGL00105: Record "GP GL00105";
+        HistGenJournalLine: Record "Hist. Gen. Journal Line";
+        OutlookSynchTypeConv: Codeunit "Outlook Synch. Type Conv";
+        InitialHistYear: Integer;
+    begin
+        InitialHistYear := GPCompanyAdditionalSettings.GetHistInitialYear();
+        if InitialHistYear > 0 then
+            GPGL20000.SetFilter(OPENYEAR, '>=%1', InitialHistYear);
+
+        if GPGL20000.FindSet() then
+            repeat
+                Clear(HistGenJournalLine);
+                if GPGL00105.Get(GPGL20000.ACTINDX) then begin
+                    HistGenJournalLine."No." := 'OpenYear-' + Format(GPGL20000.DEX_ROW_ID);
+
+#pragma warning disable AA0139
+                    HistGenJournalLine."Source Type" := ConvertSeriesToSourceType(GPGL20000.SERIES, GPGL20000.SOURCDOC.TrimEnd());
+#pragma warning restore AA0139
+
+                    HistGenJournalLine."Journal Entry No." := Format(GPGL20000.JRNENTRY);
+                    HistGenJournalLine."Account No." := CopyStr(Format(GPGL00105.ACTNUMST), 1, MaxStrLen(HistGenJournalLine."Account No."));
+                    HistGenJournalLine."Sequence No." := GPGL20000.SEQNUMBR;
+                    HistGenJournalLine."Audit Code" := GPGL20000.TRXSORCE;
+                    HistGenJournalLine.Closed := false;
+                    HistGenJournalLine.Year := GPGL20000.OPENYEAR;
+                    HistGenJournalLine."Date" := DT2Date(OutlookSynchTypeConv.LocalDT2UTC(GPGL20000.TRXDATE));
```

Comment:
I think we can rewrite many transformations here to use the data transformation. Date would need to be moved to the date time if that is applicable. Then we can get fast transfers. The disadvantage is that it would need to run within the migration, thus it would add the time to migration.